text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,210 @@
+# Meta
+[meta]: #meta
+- Name: Pack Buildpack New to Support Alternatives
+- Start Date: 2022-03-10
+- Author(s): aidandelaney
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+The `pack buildpack new` subcommand creates a new buildpack based on a shell-script template.  This proposal extends `pack buildpack new` with a `--libcnb` option and a `--template URL` option to allow alternatives to the shell-script template.  The `--libcnb` option creates a new buildpack written in golang using `libcnb`.  The `--template URL` option allows buildpack creation from an arbitrary 3rd party URL.
```

Comment:
Rather than `--libcnb`, could we use something that is Golang specific, such as `libcnb.go` or even just `go`?

Eventually, I'd like the project adopt libcnb libraries for python, bash, and other langs.

Reply:
That makes sense.  I'll make that change.

Reply:
i can't imagine we'd have a default go template that didn't include `libcnb`, so i kinda like `--golang` or similar

Reply:
Could we do without `--libcnb` or `--golang` and simply have `--template URL` where URL can be a libcnb template?

I'd personally opt-out of littering boolean flags for different possible permutations when a single parameter could  support multiple cases.

Reply:
I think I'm coming round to that view too.  Maybe `pack buildpack new [shell, golang, ...]` for buildpacks provided templates, `pack buildpack --template URL` for externally provided?

Reply:
In [RFC 0077](https://github.com/buildpacks/rfcs/blob/main/text/0077-pack-buildpack-create.md#future-work) we mention a curated repo for this. It looks like we're choosing to not do that?

Reply:
> In [RFC 0077](https://github.com/buildpacks/rfcs/blob/main/text/0077-pack-buildpack-create.md#future-work) we mention a curated repo for this. It looks like we're choosing to not do that?

@hone I'm have no strong opinion against a curated repo but I do think it's necessary to allow for _arbitrary_ urls.

> I think I'm coming round to that view too. Maybe pack buildpack new [shell, golang, ...] for buildpacks provided templates, pack buildpack --template URL for externally provided?

@AidanDelaney That's interesting. I sort of like it. :) Would it be worse if we had a single signature `pack buildpack new <name|URL>` where the first argument could be a keyword (`shell`, `golang`, etc) or URL?

Reply:
I expect to have a set of curated templates that implement our best practice.  `pack buildpack new python` would use the curated `python` scaffolding.  There is a decision around whether the curated scaffolding should be embedded or if it should reach out to a `buildpacks/templates` style repo.  `pack buildpack new python` could be a synonym for `pack buildpack new http://github.com/buildpacks/templates/python`.